tensorflow_lite_support/cc/task/vision/landmark_detector.cc

```diff
@@ -0,0 +1,157 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/cc/task/vision/landmark_detector.h"
+
+#include "absl/algorithm/container.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "flatbuffers/flatbuffers.h"
+#include "tensorflow_lite_support/cc/common.h"
+#include "tensorflow_lite_support/cc/port/integral_types.h"
+#include "tensorflow_lite_support/cc/port/status_macros.h"
+#include "tensorflow_lite_support/cc/task/core/task_api_factory.h"
+#include "tensorflow_lite_support/cc/task/core/task_utils.h"
+#include "tensorflow_lite_support/cc/task/core/tflite_engine.h"
+#include "tensorflow_lite_support/cc/task/vision/proto/landmarks_proto_inc.h"
+#include "tensorflow_lite_support/cc/task/vision/utils/frame_buffer_utils.h"
+#include "tensorflow_lite_support/metadata/cc/metadata_extractor.h"
+#include "tensorflow_lite_support/metadata/metadata_schema_generated.h"
+
+namespace tflite {
+namespace task {
+namespace vision {
+
+namespace {
+
+using ::absl::StatusCode;
+using ::tflite::metadata::ModelMetadataExtractor;
+using ::tflite::support::CreateStatusWithPayload;
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+using ::tflite::task::core::AssertAndReturnTypedTensor;
+using ::tflite::task::core::TaskAPIFactory;
+using ::tflite::task::core::TfLiteEngine;
+
+}  // namespace
+
+/* static */
+StatusOr<std::unique_ptr<LandmarkDetector>> LandmarkDetector::CreateFromOptions(
+    const LandmarkDetectorOptions& options) {
+  RETURN_IF_ERROR(SanityCheckOptions(options));
+
+  // Copy options to ensure the ExternalFile outlives the constructed object.
+  auto options_copy = absl::make_unique<LandmarkDetectorOptions>(options);
+
+  std::unique_ptr<LandmarkDetector> landmark_detector;
+  if (options_copy->base_options().has_model_file()) {
+    ASSIGN_OR_RETURN(landmark_detector,
+                     TaskAPIFactory::CreateFromBaseOptions<LandmarkDetector>(
+                         &options_copy->base_options()));
+  } else {
+    // Should never happen because of SanityCheckOptions.
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        absl::StrFormat("Missing mandatory `model_file` field in `base_options`"),
+        TfLiteSupportStatus::kInvalidArgumentError);
+  }
+  RETURN_IF_ERROR(landmark_detector->Init(std::move(options_copy)));
+
+  return landmark_detector;
+}
+
+/* static */
+absl::Status LandmarkDetector::SanityCheckOptions(
+    const LandmarkDetectorOptions& options) {
+  // Nothing to check
+  return absl::OkStatus();
+}
+/* static */
+absl::Status SanityCheckOutputTensors(

```

Comment:
This method should be invoked from Init(), not from Postprocess(). You just need to check the output tensor number once, not for every inference. 

You can use `TfLiteEngine::OutputCount(engine_->interpreter());` to get the output tensor number.